File: legacy/backend/src/main/java/app/sandori/arena/api/domain/channel/DirectChannelEntity.java
package app.sandori.arena.api.domain.channel;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "direct_channels")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class DirectChannelEntity {

    @Id
    private String channelId;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "channelId", referencedColumnName = "channelId", insertable = false, updatable = false)
    private ChannelEntity channel;

    public DirectChannelEntity(String channelId) {
        this.channelId = channelId;
    }
}

File: legacy/arena-backend/src/entities/direct-channel.entity.ts
import { Entity, JoinColumn, OneToOne, PrimaryColumn } from 'typeorm';
import { ChannelEntity } from './channel.entity';

@Entity({ name: 'direct_channels' })
export class DirectChannelEntity {
  @PrimaryColumn({ type: 'text' })
  channelId: string;

  @OneToOne(() => ChannelEntity)
  @JoinColumn({ name: 'channelId', referencedColumnName: 'channelId' })
  channel: ChannelEntity;
}
